addons/food/functions/fnc_refill.sqf

```diff
@@ -0,0 +1,47 @@
+/*
+    File Name: fnc_refill.sqf
+    Author: Jak Keen
+*/
+
+#include "../script_component.hpp"
+
+private ["_item", "_player", "_mode"];
+
+_player = _this select 1;
+_mode = _this select 2;
+
+switch (_mode) do {
+    case "bottle": {
+        {
+            if (_x in (items player)) exitwith {
+                if(count (nearestObjects [player, GVAR(refill_sources), 2]) > 0) then {
```

Comment:
The refill action is only available if you have an item to refill and if there is a refill source nearby. Why are you checking that again here?

Reply:
Fair point, I will remove